File: app/lib/Models/feedback.dart
import 'dart:convert';

Feedbacks feedbackModalFromJson(String str) => Feedbacks.fromJson(json.decode(str));

String feedbackModalToJson(Feedbacks data) => json.encode(data.toJson());

class Feedbacks {
  Feedbacks({
    this.feedbackId,
    this.comment,
    this.rating,
    this.doctorId,
    this.patientId,
    this.patientName,
    this.patientImage,
    this.doctorName,
    this.doctorImage,

  });

  String? feedbackId;
  String? comment;
  String? rating;
  String? doctorId;
  String? patientId;
  String? patientName;
  String? patientImage;
  String? doctorName;
  String? doctorImage;

  Feedbacks.fromMap(Map<dynamic, dynamic> map){
    feedbackId = map['feedback_id'];
    comment = map["comment"];
    rating = map["rating"];
    doctorId = map["doctor_id"];
    patientId = map["patient_id"];
    patientName = map["patient_name"];
    patientImage = map["patient_image"];
    doctorName = map["doctor_name"];
    doctorImage = map["doctor_image"];

  }
  factory Feedbacks.fromJson(Map<dynamic, dynamic> json) => Feedbacks(
    feedbackId: json["feedback_id"],
    comment: json["comment"],
    rating: json["rating"],
    doctorId: json["doctor_id"],
    patientId: json["patient_id"],
    patientName: json["patient_name"],
    patientImage: json["patient_image"],
    doctorName: json["doctor_name"],
    doctorImage: json["doctor_image"],
  );

  Map<String, dynamic> toJson() => {
    "feedback_id": feedbackId,
    "comment": comment,
    "rating": rating,
    "doctor_id": doctorId,
    "patient_id": patientId,
    "patient_name": patientName,
    "patient_image": patientImage,
    "doctor_name": doctorName,
    "doctor_image": doctorImage,
  };
}

File: api/models/Feedback.php
<?php

    $ds = DIRECTORY_SEPARATOR;
    $base_dir = realpath(dirname(__FILE__). $ds . '..') . $ds;

    require_once("{$base_dir}includes{$ds}Database.php"); //including Database

    class Feedback{
        private $table = 'feedbacks';

        public $Feedback_ID;
        public $Patient_ID;
        public $Doctor_ID;
        public $Rating;
        public $Remark;

        public function __construct(){
            
        }

        public function validate_params($value){
            if(!empty($value)){
                return true;
            }else{
                return false;
            }
        }

        public function add_feedback(){
            global $database;

            $this->Patient_ID = trim(htmlspecialchars(strip_tags($this->Patient_ID)));
            $this->Doctor_ID = trim(htmlspecialchars(strip_tags($this->Doctor_ID)));
            $this->Remark = trim(htmlspecialchars(strip_tags($this->Remark)));;
            $this->Rating = trim(htmlspecialchars(strip_tags($this->Rating)));

            $sql = "INSERT INTO $this->table (patient_id, doctor_id, comment, rating) VALUES (
                '" . $database->escape_value($this->Patient_ID) . "', 
                '" . $database->escape_value($this->Doctor_ID) . "', 
                '" . $database->escape_value($this->Remark) . "', 
                '" . $database->escape_value($this->Rating) . "'
            )";

            $patient_saves = $database->query($sql);

            if($patient_saves){
                return true;
            }else{
                return false;
            }
        }

        public function feedback_status(){
            global $database;

            $this->Patient_ID = trim(htmlspecialchars(strip_tags($this->Patient_ID)));
            $this->Doctor_ID = trim(htmlspecialchars(strip_tags($this->Doctor_ID)));

            $sql = "SELECT * FROM appointments WHERE 
            doctor_id = '". $database->escape_value($this->Doctor_ID) ."' AND
            patient_id = '". $database->escape_value($this->Patient_ID) ."'";

            $result_feedback = $database->query($sql);
            $result = $database->fetch_row($result_feedback);

            if(empty($result)){
                return false;
            }else{
                return true; 
            }

        }

        public function get_feedback(){
            global $database;
            $this->Doctor_ID = trim(htmlspecialchars(strip_tags($this->Doctor_ID)));

            $sql = "SELECT $this->table.patient_id as patient_id, $this->table.doctor_id as doctor_id, feedback_id, patient_name, patient_image, doctor_name, doctor_image, rating, comment FROM $this->table, patients, doctors WHERE 
            $this->table.doctor_id = '". $database->escape_value($this->Doctor_ID) ."'
            AND $this->table.doctor_id = doctors.doctor_id AND $this->table.patient_id = patients.patient_id";

            $result = $database->query($sql);
            return $database->fetch_array($result);
        
        }

        public function get_patient_feedback(){
            global $database;
            $this->Patient_ID = trim(htmlspecialchars(strip_tags($this->Patient_ID)));

            $sql = "SELECT $this->table.patient_id as patient_id, $this->table.doctor_id as doctor_id, feedback_id, patient_name, patient_image, doctor_name, doctor_image, rating, comment FROM $this->table, patients, doctors WHERE 
            $this->table.patient_id = '". $database->escape_value($this->Patient_ID) ."'
            AND $this->table.doctor_id = doctors.doctor_id AND $this->table.patient_id = patients.patient_id";

            $result = $database->query($sql);
            return $database->fetch_array($result);
        
        }

    }

    $feedback  = new Feedback();
?>